test/core.jl

```diff
@@ -35,6 +37,17 @@ for perm in ((:col, :row, :page), (:col, :page, :row),
              (:row, :page, :col), (:row, :col, :page))
     @test axisnames(permutedims(A, perm)) == perm
 end
+@test axisnames(permutedims(A, (:col,)))  == (:col, :row, :page)
```

Comment:
I added this test and the next to highlight a somewhat non-obvious feature of the `permutation` algorithm: any dimensions that aren't listed explicitly get appended, in the order in which they appear. This was useful for me at one point in Images (that's where this was effectively copied from), but I'm not sure it's necessary now and in any event I recognize it could be a bit controversial. So I wanted to air the (potentially) dirty laundry; if you don't like it, I can pull these tests and modify the algorithm to be more strict about its inputs.